Test failure regarding datetime on writing to GRIB

### What happened?

This error with Python 3.11:
```
FAILED tests/test_40_xarray_to_grib_regular_ll.py::test_canonical_dataarray_to_grib_with_grib_keys[4] - TypeError: int() argument must be a string, a bytes-like object or a real number, not 'datetime.datetime'
```


### What are the steps to reproduce the bug?

pytest

### Version

0.9.15.1

### Platform (OS and architecture)

Linux and MacOS

### Relevant log output

```shell

```

### Accompanying data

_No response_

### Organisation

ECMWF

--- a/cfgrib/cfmessage.py
+++ b/cfgrib/cfmessage.py
@@ -80,7 +80,7 @@ def to_grib_date_time(
     message, time_ns, date_key="dataDate", time_key="dataTime", epoch=DEFAULT_EPOCH
 ):
     # type: (abc.MutableField, int, str, str, datetime.datetime) -> None
-    time_s = int(time_ns) * 1e-9
+    time_s = time_ns.astype(np.int64) * 1e-9
     time = epoch + datetime.timedelta(seconds=time_s)
     datetime_iso = str(time)
     message[date_key] = int(datetime_iso[:10].replace("-", ""))

--- a/tests/test_25_cfmessage.py
+++ b/tests/test_25_cfmessage.py
@@ -19,7 +19,7 @@ def test_from_grib_date_time() -> None:
 
 def test_to_grib_date_time() -> None:
     message = {}  # type: T.Dict[str, T.Any]
-    datetime_ns = np.datetime64("2001-10-11T01:01:00", "ns").item()
+    datetime_ns = np.datetime64("2001-10-11T01:01:00", "ns")
 
     cfmessage.to_grib_date_time(message, datetime_ns)
 

